.github/workflows/ci.yml

```diff
@@ -0,0 +1,49 @@
+name: build
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  test:
+    runs-on: ${{matrix.os}}
+    timeout-minutes: 10
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-latest, ubuntu-latest, windows-latest]
+
+    steps:
+      - uses: actions/checkout@v2
+      
+      - name: Install dependencies
+        run: |
+          cd ../
+          wget https://cmake.org/files/v3.10/cmake-3.10.2-Linux-x86_64.tar.gz
+          tar -zxvf cmake-3.10.2-Linux-x86_64.tar.gz
+          wget https://github.com/google/googletest/archive/release-1.10.0.tar.gz
+          tar -zxvf release-1.10.0.tar.gz
+          ls -a
+          mkdir googletest-build
+          cd googletest-build
+          sudo ../cmake-3.10.2-Linux-x86_64/bin/cmake ../googletest-release-1.10.0
```

Comment:
I am not sure if `sudo` is allowed on Github actions CI. Can we create a conda environment and install dependencies there.

Reply:
From GitHub Docs:

```
The Linux and macOS virtual machines both run using passwordless sudo. When you need to execute commands or install tools that require more privileges than the current user, you can use sudo without needing to provide a password.
```

Ref: https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#about-virtual-environments